Create version-11-color.md (#8774)
Shopify/polaris: polaris.shopify.com/content/whats-new/version-11-color.md

---
title: Version 11 Color
description: Learn about what changes were introduced to color in Polaris v11.
imageUrl: /images/updates/new-colors-cover.png
keywords:
  - color
  - tokens
---

![Preview of new alias tokens](/images/updates/new-colors-cover.png)

## Why are we making changes?

> Over the past year we have invested a lot of time and energy into creating a more robust and cohesive token system in Polaris. We have updated typography, spacing, shadows and breakpoints. All of this work has made it easier than ever to build with Polaris and has unlocked new levels of composability within the system itself. We knew we needed to bring this same thinking to our color tokens as well.

Reviewer: This paragraph might not be needed. It's a lot of not specific information and the real value in "Why are we making these changes" is in the dot points below.

Author: I think this is nice contextual info. I get your point, but think it helps tie this work in with all the other stuff we have shipped.

Reviewer: Makes sense to me @heyjoethomas 👍

These were the main drivers for this update.

> 1. Create a defined color palette to generate tokens from. This will eliminate color sprawl and ensure a more consistent and cohesive use of color across the admin.

Reviewer: ```suggestion
1. Create a defined color palette with multiple lightness values that is used to create new tokens. This will eliminate color sprawl and ensure a more consistent and cohesive use of color across the admin.
```

We don't generate any tokens in our system. They are all chosen with strong design opinions. We use the word generate a bunch and from what I can tell they are all chosen colors that are not generated by any code or algorithms. Maybe we can use a different word? Please correct me if I am wrong.

2. Update key colors to bring more contrast and vibrancy into the admin. Key components like cards often were invisible to merchants because they had so little contrast to the app background itself. This really made it hard for merchants to quickly understand the structure of a given screen.
3. Create a consistent and thoughtful naming convention that was much simpler and more intuitive for builders to use.

> 4. Build an entirely new architecture that allows us to evolve and roll out design changes a lot easier. The previous architecture was rigid and didn’t let us easily cascade changes across the entire system.

Reviewer: The ordered list of these points is a little bit confusing to me. 2/3/4 seem a lot more important to me then 1. Maybe they should just be an unordered list?

## Color palette

> We started with the colors themselves. Doing a full audit of the colors used in Polaris. Previously we actually didn’t have a defined color palette to generate tokens from. This led to massive sprawl and inconsistency of hue, saturation, and vibrancy within our colors. We learned that there were actually 22 different gray values in Polaris, some of which were nearly identical to each other, providing no value while introducing added complexity. Every hue had a different range of tints and shades with no consistency or relationship between them.

Reviewer: ```suggestion
We started with an audit of the colors used in Polaris. Previously we didn’t have a defined color palette to create tokens from. This led to massive sprawl and inconsistency of hue, saturation, and vibrancy within our colors. We learned that there were 22 different gray values in Polaris, some of which were nearly identical to each other, providing no value while introducing added complexity. Every hue had a different range of tints and shades with no consistency or relationship between them.
```

![image of old color palette](/images/updates/old-palette.png)

> We took all these colors and generated a new color palette with 10 shades for each hue. Each shade should be meaningfully different from the ones next to it on the scale. This makes choices purposeful and clear and eliminates the sprawl and muddiness we had before.

Reviewer: ```suggestion
We took the previous colors and created a new color palette with 10 shades for each hue. Each shade should be meaningfully different from the ones next to it on the scale. This makes choices purposeful and clear and eliminates the sprawl and muddiness we had before.
```

> These colors power every token provided within the admin. This means that for the first time we can update a single color value within our palette and it will cascade down to every token referencing it, updating the entire admin from a single and simple change.

Reviewer: ```suggestion
These colors power every token provided within the admin. This means that we can update a single color value within our palette and it will cascade down to every token referencing it, updating the entire admin from a single and simple change.
```

"for the first time" makes sense and also doesn't... We could change colors through our tokens and now we are saying we can change colors from the color palette into the tokens into the admin. Maybe there is a way to frame this so it sounds like a win but also doesn't make it sound like we couldn't make color changes at scale.

...I might be reading into this too much though and this could be fine.

## Color tokens

> There are now two types of color tokens in Polaris, only one of which is actually used by builders.

Reviewer: ```suggestion
There are now two types of color tokens in Polaris. The palette tokens which are private and used to create new aliases and the alias tokens which is what designers and developers use.
```

### Palette tokens

> These tokens are not available to use directly, rather, they are where we define the color palette itself. Each color has 10 total shades.

Reviewer: I am getting stuck on this framing that the color palette is not for use... If it truely wasn't for use it would be completely private and we wouldn't talk about it. The color palette is available to be used and is documented, however there is only one way to use it and it is for creating new alias tokens.
